crates/ruff_linter/src/rules/flake8_builtins/helpers.rs

```diff
@@ -1,12 +1,16 @@
+use crate::settings::types::PythonVersion;
 use ruff_python_ast::PySourceType;
 use ruff_python_stdlib::builtins::{is_ipython_builtin, is_python_builtin};
 
 pub(super) fn shadows_builtin(
     name: &str,
-    ignorelist: &[String],
     source_type: PySourceType,
+    ignorelist: &[String],
+    python_version: PythonVersion,
 ) -> bool {
-    if is_python_builtin(name) || source_type.is_ipynb() && is_ipython_builtin(name) {
+    if is_python_builtin(name, python_version.minor())
+        || source_type.is_ipynb() && is_ipython_builtin(name)
```

Comment:
What's the use case (when checking a Jupyter notebook) for distinguishing between Python builtins and IPython builtins? Since we're refactoring the `is_python_builtin` function anyway, should we just make it a function that also accepts the `PySourceType` as well as the Python minor version? (And get rid of the `is_ipython_builtin` function?)

It seems less error-prone: currently you have to remember to call both functions if you want to know if a symbol is a builtin symbol in a notebook file

Reply:
I'll try this out.

Reply:
Oh, I think the motivation here is that the builtins check is in `uv_python_stdlib` which doesn't depend on `PySourceType`.

Reply:
Ah I see. You could probably just do it with a boolean `include_ipython_builtins` parameter... but also this doesn't need to be done in this PR!